sdcm/utils/decorators.py

```diff
@@ -155,6 +158,96 @@ def wrapped(*args, **kwargs):
     return wrapped
 
 
+class LatencyCalculatorMixedResult(GenericResultTable):
+    class Meta:
+        name = ""  # to be set by the decorator
+        description = ""
+        Columns = [
+            ColumnMetadata(name="P90 write", unit="ms", type=ResultType.FLOAT),
+            ColumnMetadata(name="P90 read", unit="ms", type=ResultType.FLOAT),
+            ColumnMetadata(name="P99 write", unit="ms", type=ResultType.FLOAT),
+            ColumnMetadata(name="P99 read", unit="ms", type=ResultType.FLOAT),
+            ColumnMetadata(name="duration", unit="HH:MM:SS", type=ResultType.DURATION)
+        ]
+
+
+class LatencyCalculatorWriteResult(GenericResultTable):
+    class Meta:
+        name = ""  # to be set by the decorator
+        description = ""
+        Columns = [
+            ColumnMetadata(name="P90 write", unit="ms", type=ResultType.FLOAT),
+            ColumnMetadata(name="P99 write", unit="ms", type=ResultType.FLOAT),
+            ColumnMetadata(name="duration", unit="HH:MM:SS", type=ResultType.DURATION)
+        ]
+
+
+class LatencyCalculatorReadResult(GenericResultTable):
+    class Meta:
+        name = ""  # to be set by the decorator
+        description = ""
+        Columns = [
+            ColumnMetadata(name="P90 read", unit="ms", type=ResultType.FLOAT),
+            ColumnMetadata(name="P99 read", unit="ms", type=ResultType.FLOAT),
+            ColumnMetadata(name="duration", unit="HH:MM:SS", type=ResultType.DURATION)
+        ]
+
+
+class ReactorStallStatsResult(GenericResultTable):
+    class Meta:
+        name = ""
+        description = ""
+        Columns = [
+            ColumnMetadata(name="total", unit="count", type=ResultType.INTEGER),
+            ColumnMetadata(name="10ms", unit="count", type=ResultType.INTEGER),
+            ColumnMetadata(name="20ms", unit="count", type=ResultType.INTEGER),
+            ColumnMetadata(name="30ms", unit="count", type=ResultType.INTEGER),
+            ColumnMetadata(name="50ms", unit="count", type=ResultType.INTEGER),
+            ColumnMetadata(name="100ms", unit="count", type=ResultType.INTEGER),
+            ColumnMetadata(name="200ms", unit="count", type=ResultType.INTEGER),
+            ColumnMetadata(name="1000ms", unit="count", type=ResultType.INTEGER),
+            ColumnMetadata(name="2000ms", unit="count", type=ResultType.INTEGER),
+        ]
+
+
+workload_to_table = {
+    "mixed": LatencyCalculatorMixedResult,
+    "write": LatencyCalculatorWriteResult,
+    "read": LatencyCalculatorReadResult
+}
+
+
+def send_result_to_argus(argus_client: ArgusClient, workload: str, name: str, description: str, cycle: int, result: dict):
+    result_table = workload_to_table[workload]()
+    result_table.name = f"{name} - {workload} - latencies"
+    result_table.description = description
+    for workload in ["write", "read"]:
```

Comment:
what about mixed ?

Reply:
latency results json contains write and read (both for mixed workload).
But this is a bug, as it overrides workload param passed to function.

Reply:
fixed to not cause confusion